src/redux/dog/reducer.tsx

```diff
@@ -0,0 +1,52 @@
+import { createAction, createReducer } from '@reduxjs/toolkit';
+
+interface DogState {
+  name: string | null;
+  sex: string;
+  age: number | null;
+  isVaccinated: boolean;
+  breed: string | null;
+  breedId: number | null;
+  isFixed: boolean;
+  bio: string | null;
+  pic: string | null;
+  // add these in once the tables are made in the BE 
+  // temperament: string | null;
+  // size: string;
+};
+
+interface Action {
+  type: string;
+  payload: any;
+};
+
+const createDog = createAction('CREATE/DOG');
+const updateDog = createAction('UPDATE/DOG');
+const getDog = createAction('GET/DOG');
+const deleteDog = createAction('DELETE/DOG');
+
+const initialState: DogState = {
+  // temperament: null,
+  // size: null,
+};
```

Comment:
const initialState: DogState = {
  // temperament: null,
  // size: null,
};
```suggestion
const initialState: DogState = {
  name: null,
  sex: 'female',
  age: null,
  isVaccinated: false,
  breed: null,
  breedId: null,
  isFixed: false,
  bio: null,
  pic: null,
  // temperament: null,
  // size: null,
};
```